Frameworks/UIKit.Xaml/ScrollView.xaml.cpp

```diff
@@ -65,6 +65,17 @@ ScrollViewer^ ScrollView::InnerScrollViewer::get() {
     return scrollViewer;
 }
 
+Windows::Foundation::Size ScrollView::ArrangeOverride(Windows::Foundation::Size finalSize) {
+    if (this->SublayerCanvas->Width < Width) {
+        // if the canvas's width is smaller than that of scrollview
+        // need adjust canvas's width to be same of the scrollview
+        // in order to have proper resizing behavior
+        this->SublayerCanvas->Width = finalSize.Width;
+    }
```

Comment:
we don't adjust the height when height is smaller because you don't want to stretch the height vertically - which is current hehavior.

---
In reply to: [108253765](https://github.com/Microsoft/WinObjC/pull/2328#discussion_r108253765) [](ancestors = 108253765)